helm-prereqs/setup-machine-a-tron.sh

```
@@ -0,0 +1,853 @@
+# NOT enough: mid-ingestion (interfaces DHCP'd, endpoints explored, machines
+# not yet created) the per-MAC creds are LIVE — purging them forces every
+# endpoint back through the credential fallback.
+if [[ "$(psql_count "SELECT count(*) FROM machines;")" == "0" \
+   && "$(psql_count "SELECT count(*) FROM machine_interfaces;")" == "0" ]]; then
+    # Batched server-side in ONE kubectl exec — thousands of entries at scale;
+    # one exec per deletion takes hours.
+    _n="$(vault_cmd 'count=0
+for m in $(vault kv list -format=yaml secrets/machines/bmc 2>/dev/null | sed "s/^- //" | grep -v "^site/"); do
+  vault kv metadata delete "secrets/machines/bmc/${m%/}/root" >/dev/null 2>&1 && count=$((count+1))
+done
+echo $count' || echo 0)"
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)